libs/api/domains/rsk/src/lib/rsk-company-info.service.ts

```diff
@@ -0,0 +1,146 @@
+import { Inject, Injectable, NotFoundException } from '@nestjs/common'
+import { RskCompany } from './models/rskCompany.model'
+import { CompanyApi } from '@island.is/clients/rsk-company-info'
+import { RskCompanyFormOfOperation } from './models/rskCompanyFormOfOperation.model'
+import { RskCompanyVat } from './models/rskCompanyVat.model'
+import { RskCompanyAddress } from './models/rskCompanyAddress.model'
+import { RskCompanyRelatedParty } from './models/rskCompanyRelatedParty.model'
+import { RskCompanyClassification } from './models/rskCompanyClassification.model'
+import { RskCompanySearchItems } from './models/rskCompanySearchItems.model'
+import { decodeBase64, toBase64 } from './rsk-company-info.utils'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+
+@Injectable()
+export class RskCompanyInfoService {
+  constructor(
+    private rskCompanyInfoApi: CompanyApi,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {}
+
+  async getCompanyInformationWithExtra(
+    nationalId: string,
+  ): Promise<RskCompany> {
+    this.logger.debug(`Service getting company by nationalId ${nationalId}`)
+    const company = await this.rskCompanyInfoApi.v1FyrirtaekjaskraSsidGet({
+      ssid: nationalId,
+    })
+    this.logger.debug(`Company in service ${company.toString()}`)
+    return {
+      nationalId: company.kennitala,
+      name: company.nafn,
+      dateOfRegistration: new Date(company.skrad),
```

Comment:
Seeing checks for dates in some places. Should they be in all?

For example line 122:
```javascript
item.skrad ? new Date(item.skrad) : undefined
```